### PR TITLE
feat(autopilot): task lifecycle — LLM-signalled completion, retention/dismiss, fix silent model-access death

### PR DESCRIPTION
**Pull request by murdore**
## Why
Autopilot "didn't work at all" and its dashboard cards "started but never completed or removed." Root-cause investigation found three real defects:

1. **Silent death (model access).** The engine defaulted `ENGINE_MODEL` to `open-fast` and `litellm-client` to `open-large` — neither is granted to this LiteLLM team (only the `*-sa` variants are). With `AUTOPILOT_MODEL` unset, every pipeline call returned *"team not allowed to access model"* → empty consensus → nothing produced, logged only as a quiet warn.
2. **No completion state.** Session status was only `idle|running|error` — a card could never show as done.
3. **No retention.** `summary-store` was insert-only; `auto-*` cards accumulated forever.

## What
**Pipeline reliability**
- Defaults → `open-fast-sa` / `open-large-sa` (accessible); a model-access 4xx now logs a clear **ERROR** instead of a silent warn.

**Task lifecycle (dashboard cards)**
- **LLM-signalled completion** folded into the existing summary call (zero extra requests) → persisted `status` + `completionReason` (new columns, idempotent ALTER migration).
- The engine gains a `completed` guard: once done it stops running pipelines / re-pushing, until a `tool-started` event re-arms it. The driver halts injection on a completed terminal and clears on fresh WS work. A one-shot **"✅ done"** push fires (not presence-gated, so it never gets lost).
- **Retention:** `pruneOld(maxAgeDays + per-terminal cap)` at startup + hourly (completed cards exempt from the cap; NULL-terminal rows partitioned independently), plus `DELETE /api/summaries` and a dismiss ✕ button on each card.

## Review
An adversarial review pass surfaced 6 real issues (repeated completion pushes, driver halt-not-clearing, prune deleting the completed card, NULL-partition lumping, lost completion push, missing length caps) — **all fixed and re-verified** in this commit.

## Verification
- `pnpm lint --max-warnings 0`, `svelte-check` + `tsc --strict`: 0 errors.
- Full test suite green, incl. 8 new `tests/summary-store.test.cjs` (columns, migration, age/cap pruning, completed-exemption, NULL partitions, deleteById).
- Runtime: built + booted (log shows `model=open-fast-sa`); drove `POST/GET/DELETE /api/summaries` — `status`/`completionReason` round-trip, `DELETE`→`{removed:1}`→404, oversized `id`/`completionReason`→400.
- Not driven live: the end-to-end engine→LLM→completed→halt loop (needs a running managed agent terminal).

Single commit, rebased on `release` (single-commit-policy compliant).

## Summary by CodeRabbit

- **New Features**
  - Session summaries now display completion state, including a completion reason.
  - Dashboard summary cards can be dismissed individually.
- **Bug Fixes**
  - Autopilot stops re-running for completed sessions until new activity occurs.
  - Summary retention/pruning is improved, including correct handling of completed items.
  - Dismissal uses optimistic removal and safely restores cards if deletion fails.
- **Tests**
  - Added a test suite covering summary-store migration, persistence, retention, and deletion behavior.

### CODE REVIEW

File: src/lib/modules/server/sessions/litellm-client.ts [hunk 1]
```diff
@@ -134,9 +134,19 @@ async function attemptCompletion<T>(
       // produce empty consensus invisibly (no log, no signal). 429 = the LiteLLM key's
       // max_parallel_requests is exhausted (often by other processes sharing the key).
       const detail = await res.text().catch(() => '');
-      console.warn(
-        `[litellm] HTTP ${res.status} (model=${body.model as string}, jsonMode=${jsonMode}): ${detail.slice(0, 200)}`
-      );
+      // A model-access denial is a CONFIGURATION error, not a transient blip: the configured model
```
Comment (Tara-ag) [inline src/lib/modules/server/sessions/litellm-client.ts:139]: 💡 **SUGGESTION: Model access error regex could be more precise**

The regex `/not allowed to access model|can only access models/i` is good but might miss variations like "access denied for model" or "model access forbidden". Consider expanding it:

```typescript
const isModelAccess = /(?:not allowed|cannot|unable|forbidden|denied).*?(?:access|use).*?model|model.*?(?:not available|restricted|disabled)/i.test(detail);
```

Also, the error message includes the raw `detail` slice which could contain sensitive information from the LiteLLM gateway. While this is logged server-side (not exposed to clients), ensure your log aggregation doesn't inadvertently expose API keys or internal routing details that might appear in the error response.

File: tests/summary-store.test.cjs [hunk 2]
```diff
@@ -0,0 +1,221 @@
+/**
```
Comment (Tara-ag) [inline tests/summary-store.test.cjs:1]: ✅ **Good: Comprehensive test coverage**

The test file covers:
- Round-trip persistence of new columns
- Migration from old schema
- Age-based pruning
- Per-terminal cap pruning with completed exemption
- NULL terminal partitioning
- deleteById functionality

The use of isolated temp directories per test is the correct pattern (following `DeviceTokenStore` idiom). Tests exercise the real TS module via tsx/cjs which gives confidence the implementation works.

**Minor suggestion:** Consider adding a test for the edge case where `maxAgeDays` is 0 (should delete all rows) and negative values (should be handled gracefully).

File: src/lib/modules/client/dashboard/AutopilotPanel.svelte [hunk 3]
```diff
@@ -61,6 +61,25 @@
     }
   }
 
+  async function dismiss(id: string): Promise<void> {
```
Comment (Tara-ag) [inline src/lib/modules/client/dashboard/AutopilotPanel.svelte:69]: 💡 **SUGGESTION: Optimistic dismissal could lose user data on network failure**

The optimistic UI pattern here (removing the card before server confirmation) is good for perceived performance, but there's a subtle UX issue: if the DELETE request fails with a 5xx error, the card is restored, but any user interaction with other cards during the network round-trip could be disrupted.

Consider adding a loading state instead:

```typescript
async function dismiss(id: string): Promise<void> {
  const idx = summaries.findIndex((s) => s.id === id);
  if (idx === -1) return;
  
  // Mark as dismissing instead of removing
  summaries[idx] = { ...summaries[idx], _dismissing: true };
  
  try {
    const res = await fetch('/api/summaries', { ... });
    if (res.ok || res.status === 404) {
      summaries = summaries.filter((s) => s.id !== id);
    } else {
      delete summaries[idx]._dismissing; // Restore
    }
  } catch {
    delete summaries[idx]._dismissing; // Restore
  }
}
```

And disable the dismiss button while `_dismissing` is true.

File: src/routes/api/summaries/+server.ts [hunk 4]
```diff
@@ -78,6 +83,39 @@ export const POST: RequestHandler = async ({ request }) => {
   return json({ id: record.id, success: true }, { status: 201 });
 };
 
+export const DELETE: RequestHandler = async ({ request }) => {
```
Comment (Tara-ag) [inline src/routes/api/summaries/+server.ts:86]: ✅ **Good: Proper input validation on DELETE endpoint**

The validation here follows the established patterns:
- Auth check first
- JSON parsing with try/catch
- Type checking for `id`
- Length cap (128 chars) before DB interaction
- 404 for not found (idempotent-ish)

This aligns with the project's security requirements for input validation at the HTTP boundary.

File: src/lib/modules/server/sessions/autopilot-engine.ts [hunk 5]
```diff
@@ -155,13 +168,15 @@ export function startAutopilotEngine(): void {
   } catch {
```
Comment (Tara-ag) [inline src/lib/modules/server/sessions/autopilot-engine.ts:169]: 💡 **MINOR: Timer reference not stored, prevents graceful shutdown cleanup**

The `setInterval` for pruning returns a timer handle that's immediately unreferenced with `.unref()`, but the handle itself isn't stored anywhere. If the engine needs to be stopped cleanly (e.g., in tests or during server shutdown), this interval cannot be cleared.

**Suggested fix:** Store the timer handle and clean it up in a shutdown function:

```typescript
let pruneTimer: ReturnType<typeof setInterval> | null = null;

// In startAutopilotEngine():
pruneTimer = setInterval(pruneSummaries, PRUNE_INTERVAL_MS);
pruneTimer.unref();

// Add to shutdown/unsubscribe logic:
if (pruneTimer) {
  clearInterval(pruneTimer);
  pruneTimer = null;
}
```

File: src/lib/modules/server/sessions/summary-store.ts [hunk 6]
```diff
@@ -87,17 +99,77 @@ export class SummaryStore {
       .all(limit) as Record<string, unknown>[];
     return rows.map(rowToRecord);
   }
+
```
Comment (Tara-ag) [inline src/lib/modules/server/sessions/summary-store.ts:103]: 💡 **MINOR: Potential SQL injection via ROW_NUMBER window function**

While the query uses parameterized statements for values, the `ORDER BY` clause inside the `ROW_NUMBER() OVER` window function concatenates column names directly. Although `maxPerTerminal` is validated as a number, if this code is ever modified to accept dynamic column ordering, it could introduce SQL injection.

**Current code is safe** (columns are hardcoded), but consider adding a comment warning future maintainers:

```typescript
// SECURITY: Column names in ORDER BY are hardcoded. Never interpolate user input here.
```

Alternatively, validate `maxPerTerminal` is a positive integer at the function boundary to be extra defensive.

File: src/lib/modules/server/sessions/autopilot-engine.ts [hunk 7]
```diff
@@ -433,11 +525,24 @@ async function runPipeline(
     if (!enabled || session.cancelled) {
```
Comment (Tara-ag) [inline src/lib/modules/server/sessions/autopilot-engine.ts:551]: ⚠️ **MAJOR: Race condition in completion push**

The `pushCompletion` function is called with `await` inside `runPipeline`, but there's a race condition window:

1. Line 525-530: `session.completed = true` is set AFTER `persist()` but BEFORE `await pushCompletion()`
2. If a concurrent event arrives between these two operations, `handleEvent` (line 272-274) will see `session.completed === false` and allow another pipeline run
3. This could result in duplicate completion notifications

**Suggested fix:** Set `session.completed = true` atomically before calling `persist()`, or use a proper lock mechanism. Option 1: Mark complete before persist so the guard in handleEvent catches concurrent events immediately.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 9 changed files
**New issues raised:** 6 (1 MAJOR, 2 MINOR, 3 SUGGESTION/positive)

### Issues Found

| Severity | Count | Description |
|----------|-------|-------------|
| 🔒 CRITICAL | 0 | None found |
| ⚠️ MAJOR | 1 | Race condition in completion push timing |
| 💡 MINOR | 2 | Timer cleanup, SQL comment suggestions |
| 💬 SUGGESTION | 3 | UX improvement, regex refinement, test coverage |

### Key Observations

**Security:** ✅ No blocking security issues. The PR properly:
- Validates all inputs at HTTP boundaries with length caps
- Uses `validateAuth()` middleware on all new endpoints
- Parameterizes SQL queries (no injection vectors)
- Doesn't log or expose secrets

**Correctness:** ⚠️ One race condition identified in `autopilot-engine.ts` where `session.completed` is set after `persist()` but before `await pushCompletion()`, potentially allowing duplicate completion notifications under concurrent load.

**Testing:** ✅ Excellent coverage with 8 new tests covering:
- Column migration from old schema
- Age-based and per-terminal pruning
- Completed-row exemption logic
- NULL terminal partitioning
- deleteById functionality

**Standards Compliance:** ✅ Follows project patterns:
- Types properly added to `src/lib/types/autopilot.ts`
- Barrel export used correctly
- Input validation matches existing patterns (128-char ID cap, 500-char completionReason cap)
- SQLite WAL mode with proper cleanup

### Recommendation

**Approve** - The race condition is a real bug but low-impact (at worst, a duplicate push notification). The PR addresses the core issues described (silent model-access death, no completion state, unbounded retention) and includes comprehensive tests. The race condition can be addressed in a follow-up or fixed before merge at author's discretion.

File: src/lib/modules/server/sessions/litellm-client.ts (hunk 1 above)
Comment (murdore) [reply src/lib/modules/server/sessions/litellm-client.ts:139]: Addressed in `97b09c7`: added a clarifying comment — `detail` is the gateway's error *response* body (it carries no `Authorization`/API key), and the regex is deliberately narrow to the two observed LiteLLM substrings so a transient/validation 4xx isn't misclassified as a permanent config error. Behavior unchanged. Thanks @Tara-ag.

File: tests/summary-store.test.cjs (hunk 2 above)
Comment (murdore) [reply tests/summary-store.test.cjs:1]: Done in `97b09c7`: added `pruneOld` tests for `maxAgeDays: 0` (deletes every row before `now`) and a negative value (cutoff pushed into the future, no throw). Thanks @Tara-ag.

File: src/lib/modules/client/dashboard/AutopilotPanel.svelte (hunk 3 above)
Comment (murdore) [reply src/lib/modules/client/dashboard/AutopilotPanel.svelte:69]: Fixed in `97b09c7`: `dismiss()` no longer snapshots/restores the whole `summaries` array — it captures only the removed record and reinserts just that one (re-sorted newest-first) on failure, so a concurrent dismissal or a fresher 8s poll can't be clobbered. Thanks @Tara-ag.

File: src/routes/api/summaries/+server.ts (hunk 4 above)
Comment (murdore) [reply src/routes/api/summaries/+server.ts:86]: Thanks @Tara-ag — no change needed; the validation order (auth → JSON parse → id type → 128-char cap → `deleteById` → 404) is intentional and matches the project pattern.

File: src/lib/modules/server/sessions/autopilot-engine.ts (hunk 5 above)
Comment (murdore) [reply src/lib/modules/server/sessions/autopilot-engine.ts:169]: Fixed in `97b09c7`: the interval handle is now stored (`pruneTimer`) and a new exported `stopAutopilotEngine()` clears it (plus the event `unsubscribe`), wired into `server.ts`'s `shutdown()` alongside the other teardowns. Thanks @Tara-ag.

File: src/lib/modules/server/sessions/summary-store.ts (hunk 6 above)
Comment (murdore) [reply src/lib/modules/server/sessions/summary-store.ts:103]: Addressed in `97b09c7`: added a SECURITY comment above the window query. The `PARTITION BY`/`ORDER BY` identifiers are hardcoded literals; the only dynamic value (`maxPerTerminal`) is bound as a `?` parameter, never string-concatenated. Thanks @Tara-ag.

File: src/lib/modules/server/sessions/autopilot-engine.ts (hunk 7 above)
Comment (murdore) [reply src/lib/modules/server/sessions/autopilot-engine.ts:551]: No change needed, @Tara-ag: `persist()` is synchronous (`summaryStore.insert()` is a plain better-sqlite3 `.run()`), so there is no `await` between it and `session.completed = true` — JS run-to-completion means no `handleEvent` can interleave in that span. Independently, `session.running` stays `true` for the entire pipeline (including the `pushCompletion` await), so a concurrent pipeline run is already blocked regardless of where `completed` is set.

File: src/lib/modules/client/dashboard/autopilot-driver.svelte.ts [hunk 8]
```diff
@@ -320,6 +323,10 @@ export class AutopilotDriver {
         }
         latest.add(tid);
         rt.consensus = { agentCount: 5, quorum: 3, steps: parseSteps(rec.nextSteps) };
+        // The latest card's status gates the loop: once the engine marks the task complete the
+        // driver stops injecting. It self-heals — newer active summaries (fresh work in the same
+        // terminal) flip this back to false on a later poll.
+        rt.completed = rec.status === 'completed';
```
[the remark this answers is not in the document]
Comment (murdore) [reply src/lib/modules/client/dashboard/autopilot-driver.svelte.ts]: Fixed in `97b09c7`, @coderabbitai — both points. (1) `fetchSummaries()` now freshness-gates the completed write: `rt.completed = true` only when `createdAt >= rt.lastEventAt`, so a stale completed row can't re-halt a terminal that a `tool-started` event just self-healed. (2) Narrowed the WS re-arm in `handleEvent` to `tool-started` only, matching the engine's `session.completed` re-arm contract.

File: src/lib/modules/client/dashboard/AutopilotPanel.svelte [hunk 9]
```diff
@@ -61,6 +61,25 @@
...
+      });
+      // 404 = already gone (fine). Any other failure → restore so the user can retry.
+      if (!res.ok && res.status !== 404) {
+        summaries = prev;
+      }
+    } catch {
+      summaries = prev;
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (murdore) [reply src/lib/modules/client/dashboard/AutopilotPanel.svelte:101]: Fixed in `97b09c7`, @coderabbitai: `dismiss()` captures only the removed record and reinserts just that one on failure (re-sorted newest-first) instead of restoring a stale whole-array snapshot, so concurrent dismissals / fresher polls aren't clobbered.

File: src/lib/modules/server/sessions/summary-store.ts [hunk 10]
```diff
@@ -32,29 +27,44 @@ const COLUMNS = [
...
+  'status',
+  'completion_reason',
 ] as const;
 
-// ── Row ↔ Record conversion ──────────────────────────────────────────
+const DAY_MS = 24 * 60 * 60 * 1000;
+
+// ── SummaryStore ─────────────────────────────────────────────────────
 
 export class SummaryStore {
```
[the remark this answers is not in the document]
Comment (murdore) [reply src/lib/modules/server/sessions/summary-store.ts:39]: Done in `97b09c7`, @coderabbitai: added `/** SQLite-backed persistence + retention for autopilot session-summary records. */` directly above `export class SummaryStore`.

File: src/routes/api/summaries/+server.ts [hunk 11]
```diff
@@ -78,6 +83,39 @@ export const POST: RequestHandler = async ({ request }) => {
...
+  }
+
+  const removed = summaryStore.deleteById(id.trim());
+  if (removed === 0) {
+    // Unknown / already-dismissed id → 404 so the caller can tell "gone" from a real delete.
+    return json({ error: 'Summary not found', id: id.trim() }, { status: 404 });
+  }
+  return json({ removed, success: true });
+};
+
```
[the remark this answers is not in the document]
Comment (murdore) [reply src/routes/api/summaries/+server.ts:124]: Fixed in `97b09c7`, @coderabbitai: wrapped `summaryStore.deleteById()` in try/catch returning a JSON `500`, mirroring the POST insert handler's error path.

**Comment by murdore** [comment]
## Review comments addressed — `97b09c7`

Thanks @Tara-ag and @coderabbitai. Triaged all 11 inline comments and replied in-thread. Summary:

**Fixed (4 real issues):**
- **Stale `completed` re-halt** (driver) — the poll now freshness-gates `rt.completed` (`createdAt >= lastEventAt`); WS re-arm narrowed to `tool-started` to match the engine's contract.
- **`dismiss()` whole-array restore** — now restores only the single removed record, so a concurrent dismissal / fresher poll isn't clobbered.
- **Prune-timer teardown** — handle stored + new `stopAutopilotEngine()` wired into `server.ts` shutdown.
- **DELETE error handling** — `deleteById()` wrapped in try/catch → JSON 500.

**Nits done:** `pruneOld` edge-case tests (0 / negative), SECURITY comment on the window query, `SummaryStore` JSDoc, and a litellm log clarification.

**No change (verified false alarms):** the `persist()`→`completed` "race" (synchronous persist + `session.running` guards the whole pipeline) and the approving note on DELETE validation.

Verified: lint `--max-warnings 0`, `tsc`/`svelte-check` 0 errors, full test suite (10 store tests), runtime DELETE + graceful shutdown. Kept as a single commit.

File: src/lib/modules/server/sessions/summary-store.ts [hunk 12]
```diff
@@ -87,17 +100,80 @@ export class SummaryStore {
       .all(limit) as Record<string, unknown>[];
     return rows.map(rowToRecord);
   }
+
+  /**
+   * Retention. Deletes records older than `maxAgeDays`, then trims each
+   * terminal's history to its newest `maxPerTerminal` rows. Either bound is
+   * optional; returns the total number of rows removed. `now` is injectable
+   * for deterministic tests.
```
Comment (Tara-ag) [inline src/lib/modules/server/sessions/summary-store.ts:108]: 💡 **MINOR: maxPerTerminal lacks validation for positive integer and upper bounds**

The `pruneOld` function accepts `maxPerTerminal` without validating it's a positive integer or enforcing reasonable upper bounds. While the SQL uses parameterized queries (safe from injection), negative/zero values could cause unintended deletion behavior, and extremely large values could impact performance.

**Suggested fix:** Add validation at the function boundary:

```typescript
if (opts?.maxPerTerminal !== undefined) {
  const maxPerTerm = Math.max(0, Math.min(opts.maxPerTerminal, 1000));
  if (maxPerTerm > 0) {
    // ... existing query logic using maxPerTerm ...
  }
}
```

This mirrors the defensive pattern used elsewhere in the codebase and prevents edge cases from causing unexpected behavior.

File: src/lib/modules/server/sessions/autopilot-engine.ts [hunk 13]
```diff
@@ -373,6 +429,47 @@ async function push(session: EngineSession, summary: string, steps: NextStep[]):
...
+ * completion fires exactly once (the session.completed guard prevents re-fires), so we always
+ * deliver it — otherwise a task that finishes while the dashboard is open, then the viewer leaves
+ * with no further events, would never notify at all.
+ */
+async function pushCompletion(
+  session: EngineSession,
+  summary: string,
+  reason: null | string
+): Promise<void> {
+  const port = process.env.PORT || '54007';
```
Comment (Tara-ag) [inline src/lib/modules/server/sessions/autopilot-engine.ts]: 💬 **SUGGESTION: Hardcoded localhost may fail if server binds to different interface**

Both `push()` and `pushCompletion()` use hardcoded `http://127.0.0.1:${port}/api/notify`. If the server is configured to bind to a specific IP address or `0.0.0.0`, this internal fetch could fail.

**Consider:** Using a configurable host or documenting that the server must be accessible via localhost for autopilot notifications to work. This is a minor concern since the default configuration works correctly.

File: src/lib/modules/client/dashboard/AutopilotPanel.svelte [hunk 14]
```diff
@@ -61,6 +61,33 @@
     }
   }
 
+  async function dismiss(id: string): Promise<void> {
```
Comment (Tara-ag) [inline src/lib/modules/client/dashboard/AutopilotPanel.svelte:69]: 💡 **MINOR: Double-click on dismiss button can spawn multiple concurrent DELETE requests**

The `dismiss()` function has no loading state or disabled button state during the async operation. While the `restore()` function has a duplicate-guard check (`!summaries.some((s) => s.id === id)`), there's no prevention of concurrent DELETE requests being sent to the server.

**Suggested fix:** Track in-flight dismiss operations and disable the button:

```typescript
// Add to component state
let dismissingIds = new Set<string>();

async function dismiss(id: string): Promise<void> {
  if (dismissingIds.has(id)) return;
  dismissingIds.add(id);
  dismissingIds = dismissingIds; // trigger reactivity
  
  // ... existing logic ...
  
  dismissingIds.delete(id);
  dismissingIds = dismissingIds;
}

// In template:
<button
  class="dismiss-btn"
  disabled={dismissingIds.has(s.id)}
  ...
>
  {dismissingIds.has(s.id) ? '...' : '✕'}
</button>
```

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 10 changed files
**New issues raised:** 3 (all MINOR/SUGGESTION)
**Existing comments:** 11 (from prior reviewers)

### Analysis

This PR implements the autopilot task lifecycle feature with LLM-signalled completion, retention/pruning, and dashboard card dismissal. The implementation is solid and addresses the root causes described in the PR description.

### Security Assessment ✅

- **No hardcoded secrets** - API keys read from environment, never committed
- **No command injection** - All SQL uses parameterized queries
- **No auth bypass** - All endpoints validate bearer tokens
- **Secret handling** - API_KEY used in headers but never logged
- **Input validation** - Length caps (128 char ID, 500 char completionReason), type checking present

### Issues Raised (Non-blocking)

1. **MINOR:** Double-click on dismiss button can spawn concurrent DELETE requests (no loading state)
2. **MINOR:** `maxPerTerminal` lacks positive integer and upper bound validation
3. **SUGGESTION:** Hardcoded localhost in push functions may fail with non-default bind configurations

### Existing Review Comments

Prior reviewers (Tara-ag, CodeRabbit) raised 11 points covering:
- Model access error regex precision (SUGGESTION)
- Timer reference cleanup (MINOR - addressed)
- SQL injection defense comment (MINOR)
- Race condition in completion push (MAJOR - code correctly handles this)
- Stale summary handling (MAJOR - addressed)
- Optimistic dismissal restoration (MAJOR - addressed)
- JSDoc comments (MINOR - addressed)
- DELETE error handling (MINOR - addressed)

### Verification

- Single commit policy: ✅ Compliant (1 commit)
- Type safety: ✅ New types in `src/lib/types/autopilot.ts` (not generated/)
- Tests: ✅ 8 new tests in `tests/summary-store.test.cjs`
- CI checks: All passing (lint, format, check, build, tests)

**Decision:** Approve - No blocking issues found. The 3 new MINOR/SUGGESTION items can be addressed in follow-up or accepted as-is.

File: src/lib/modules/server/sessions/summary-store.ts (hunk 12 above)
Comment (murdore) [reply src/lib/modules/server/sessions/summary-store.ts:108]: Fixed in `824650b`: clamped `maxPerTerminal` to an integer in [0, 1000] (`Math.trunc` + `Math.min`/`Math.max`). A 0/negative cap is now a **no-op** (the DELETE is skipped) instead of matching `rn > 0` and deleting every active row, and absurdly large values are bounded. Added a unit test covering 0 and negative. Thanks @Tara-ag.

File: src/lib/modules/server/sessions/autopilot-engine.ts (hunk 13 above)
Comment (murdore) [reply src/lib/modules/server/sessions/autopilot-engine.ts]: Addressed in `824650b`: extracted `internalNotifyUrl()`, used by both `push()` and `pushCompletion()`. Host defaults to `127.0.0.1` (the server binds all interfaces — `server.listen(port)` with no host — so loopback always resolves for this same-process call) and is now overridable via `SHOOTER_INTERNAL_HOST` for setups that pin the server to a specific interface and block loopback. Thanks @Tara-ag.

File: src/lib/modules/client/dashboard/AutopilotPanel.svelte (hunk 14 above)
Comment (murdore) [reply src/lib/modules/client/dashboard/AutopilotPanel.svelte:69]: Fixed in `824650b`: added a `SvelteSet` in-flight guard — `dismiss()` early-returns if the id is already being deleted, cleaned up in a `finally`, and the button is `disabled` (dimmed) while its DELETE is in flight. A double-click can no longer spawn concurrent DELETE requests. Thanks @Tara-ag.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 10  
**New issues found:** 0  
**Status:** ✅ Approved

### Verification

All previously identified issues have been addressed by the author:

| Issue | Status | Resolution |
|-------|--------|------------|
| Race condition in `dismiss()` | ✅ Fixed | Single-record capture instead of whole-array snapshot |
| Stale completed summary override | ✅ Fixed | Freshness-gated with `createdMs >= rt.lastEventAt` |
| Timer reference not stored | ✅ Fixed | `pruneTimer` stored and cleared in `stopAutopilotEngine()` |
| DELETE error handling | ✅ Fixed | Try/catch wrapper with JSON 500 response |
| `maxPerTerminal` validation | ✅ Fixed | Clamped to [0, 1000] range |
| Double-click dismiss guard | ✅ Fixed | `SvelteSet` in-flight tracking + disabled state |
| JSDoc for `SummaryStore` | ✅ Fixed | Added concise class documentation |
| SQL injection warning | ✅ Fixed | SECURITY comment added |

### Security Checklist

- ✅ Auth on DELETE endpoint (`validateAuth` first)
- ✅ Input validation (id ≤128 chars, completionReason ≤500 chars)
- ✅ Parameterized SQL queries (no string interpolation)
- ✅ No secrets in logs or error messages
- ✅ Resource cleanup on shutdown (`stopAutopilotEngine` wired in)

### Test Coverage

The new `tests/summary-store.test.cjs` covers:
- Round-trip persistence of new columns
- Schema migration from old DBs
- Age-based pruning (including edge cases: 0, negative)
- Per-terminal cap pruning with completed exemption
- NULL-terminal partitioning
- `deleteById` functionality

The PR successfully fixes the three root causes:
1. **Silent model-access death** → Now logs ERROR with actionable hint
2. **No completion state** → Added `status` + `completionReason` with LLM-signalled completion
3. **No retention** → Added `pruneOld()` with startup + hourly pruning, plus dismiss UI

Single commit, rebased on `release` — compliant with repo policy.
